kubernetes-client/src/main/java/io/fabric8/kubernetes/client/http/HttpRequest.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.kubernetes.client.http;
+
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public interface HttpRequest extends HttpHeaders {
+  
+  public interface Builder extends BasicBuilder {
+
+    HttpRequest build();
+
+    Builder uri(String uri);
+
+    Builder url(URL url);
+    
+    @Override
+    Builder uri(URI uri);
+
+    default Builder put(String contentType, String writeValueAsString) {
+      return method("PUT", contentType, writeValueAsString);
+    }
+
+    default Builder post(String contentType, String writeValueAsString) {
+      return method("POST", contentType, writeValueAsString);
+    }
+    
+    Builder post(String contentType, byte[] writeValueAsBytes);
+    
+    Builder post(String contentType, InputStream stream, long length);
+
+    default Builder delete(String contentType, String writeValueAsString) {
```

Comment:
Out of curiosity, are we sure we want/need to provide a delete with body method?
Is any of the Kubernetes HTTP endpoints actually expecting requests like this one?

Reply:
arrg. OK, delete options go in the body of the request.

Reply:
https://github.com/kubernetes/kubernetes/issues/34856